Automattic/simplenote-macos
# Fastlane: Finalize Release

### Description:
In this PR we're enabling support for localizations in Fastlane's `finalize_release` step. This means that:
- We'll download Metadata (AppStore Strings!)
- We'll also download translated Localizable.strings
- And as per PR #605, we'll **also** download the Main Menu subproject translations.

Ref. #8

/cc @mokagio @loremattei 
Thanks in advance gentlemen!!

### Details:
- `VERSION_LONG` has been patched to match [WooCommerce style](https://github.com/woocommerce/woocommerce-ios/blob/develop/config/Version.Public.xcconfig#L4). This was causing issues when updating the `Metadata` GlotPress Key.
- Fastlane: `finalize release` now runs `ios_update_metadata` as well!
- We're bringing [download_metadata.swift](https://github.com/Automattic/simplenote-ios/blob/develop/fastlane/download_metadata.swift) from Simplenote iOS (with its URL patched!)

### Test
Testing this PR is quite tricky, because the actual scritps perform **git push**. 
For that reason, we'll need to disable few things in your local checkout. **Please:**

1. Locate **ios_git_helper.rb** in your local checkout
   - Comment [this commit OP](https://github.com/wordpress-mobile/release-toolkit/blob/bb668d63f550cb50f675470ee6a7b03ed8fb5419/lib/fastlane/plugin/wpmreleasetoolkit/helper/ios/ios_git_helper.rb#L99) 
   - Comment [this other commit OP](https://github.com/wordpress-mobile/release-toolkit/blob/bb668d63f550cb50f675470ee6a7b03ed8fb5419/lib/fastlane/plugin/wpmreleasetoolkit/helper/ios/ios_git_helper.rb#L103)
   - Comment [this push OP](https://github.com/wordpress-mobile/release-toolkit/blob/bb668d63f550cb50f675470ee6a7b03ed8fb5419/lib/fastlane/plugin/wpmreleasetoolkit/helper/ios/ios_git_helper.rb#L105)
2. Open your `fastlane/Fastfile`
   - Comment [this line](https://github.com/Automattic/simplenote-macos/pull/612/files#diff-56c5021ef06ac6cea807ae53269b6f4fR148) (disables `finalize_prechecks`!)
   - Comment [this line](https://github.com/Automattic/simplenote-macos/pull/612/files#diff-56c5021ef06ac6cea807ae53269b6f4fR150) (disables `ios_final_tag`)
3. **Turn off your wifi!!!**
4. Run `bundle exec fastlane finalize_release`

- [ ] Verify the following files were updated in your local repository, for every language:
  - [ ] **keywords.txt**
  - [ ] **description.txt**
  - [ ] **promotional.txt**
  - [ ] **release.txt**
  - [ ] **Localizable.strings**
  - [ ] **MainMenu.strings**

### Release
These changes do not require release notes.
